Makefile

```diff
@@ -7,12 +7,14 @@ SKOPEO_BRANCH = master
 SUDO =
 
 ifeq ($(shell uname),Darwin)
-DARWIN_BUILD_TAG = containers_image_ostree_stub
+	DARWIN_BUILD_TAG = containers_image_ostree_stub
 endif
 BUILDTAGS = btrfs_noversion libdm_no_deferred_remove $(DARWIN_BUILD_TAG)
 BUILDFLAGS := -tags "$(BUILDTAGS)"
 
 PACKAGES := $(shell go list $(BUILDFLAGS) ./... | grep -v github.com/containers/image/vendor)
+SOURCE_DIRS := $(shell echo $(PACKAGES) | awk 'BEGIN{FS="/"; RS=" "}{print $$4}' | uniq)
```

Comment:
If this is used in only one rule (unlike `PACKAGES`), and not intended to be overwritten by the user, it would be better to compute the list inside that single rule so that we avoid the three extra processes on every `make`.

Reply:
True, but I think the overhead is minimal and we would use consistently simply expanded variables. The difference is really subtle (in this case) and I think many people are simply not aware of the difference between ':=' and '='.

If you feel strongly about it, I can change it.

Reply:
`=` vs. `:=` would minimize the run-time overhead, but it would still mean a global variable with a single user.  The `make` variable can be entirely eliminated by computing `$source_dirs` inside the shell rules for `fmt`.

Reply:
> = vs. := would minimize the run-time overhead, but it would still mean a global variable with a single user. 

Right. IM, however,  it makes sense to define it globally. Partly because it creates some grouping which helps me understand the build and partly since I see this as a valuable variable for future targets.

> The make variable can be entirely eliminated by computing $source_dirs inside the shell rules for fmt.

Sure

What's about using '=' then?

Reply:
> Partly because it creates some grouping which helps me understand the build

OTOH it splits the working of the `fmt` target into two faraway places.

> since I see this as a valuable variable for future targets.

Maybe; an uncertain bet.

> What's about using '=' then?

Fair enough, better than `:=` at least.